test/new-e2e/tests/agent-subcommands/dogstatsdreplay/dogstatsdreplay_common_test.go

```diff
@@ -0,0 +1,72 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package dogstatsdreplay contains e2e tests for the dogstatsd-replay command
+package dogstatsdreplay
+
+import (
+	_ "embed"
+	"encoding/base64"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/e2e"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/environments"
+)
+
+//go:embed fixtures/metrics_capture.zstd
+var metricsWithTagsCapture []byte
+
+type baseDogstatsdReplaySuite struct {
+	e2e.BaseSuite[environments.Host]
+}
+
+// uploadCaptureFile uploads a capture file to the remote host
+func (v *baseDogstatsdReplaySuite) uploadCaptureFile(captureData []byte, remotePath string) {
+	encoded := base64.StdEncoding.EncodeToString(captureData)
```

Comment:
I think it would make sense to use `Host::WriteFile` (from `new-e2e/pkg/utils/e2e/client/host.go`) here.

Reply:
I don't think `WriteFile` works here because the capture data is compressed in zstd format which is then interpreted into unintended characters. Base64 encoding it before echo is safer because of the compression.